shelley/chain-and-ledger/shelley-spec-ledger-test/src/Test/Shelley/Spec/Ledger/ApplyBlock.hs

```diff
@@ -0,0 +1,195 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+{-# OPTIONS_GHC -Wno-orphans #-}
+
+module Test.Shelley.Spec.Ledger.ApplyBlock where
+
+import Control.Monad.State
+import Shelley.Spec.Ledger.BaseTypes
+import Data.Proxy
+import Data.Traversable
+import qualified Data.Map as Map
+import qualified Data.Set as Set
+import qualified Data.Sequence.Strict as StrictSeq
+import Data.Foldable
+import Data.Semigroup (Max(..))
+
+import Cardano.Ledger.Shelley (ShelleyEra)
+import Shelley.Spec.Ledger.API.Protocol (PraosCrypto)
+import Shelley.Spec.Ledger.API.Validation
+import Shelley.Spec.Ledger.Address
+import Cardano.Ledger.SafeHash
+import Cardano.Ledger.Coin (Coin (..))
+import Shelley.Spec.Ledger.TxBody
+import Shelley.Spec.Ledger.LedgerState
+import qualified Cardano.Crypto.Hash.Class as CHC
+import Cardano.Crypto.Util (SignableRepresentation)
+import qualified Cardano.Crypto.KES.Class as KES
+import Cardano.Ledger.Crypto (KES, DSIGN)
+import qualified Shelley.Spec.Ledger.UTxO as UTxO
+import qualified Cardano.Crypto.DSIGN.Class as DSIGN
+import Shelley.Spec.Ledger.Keys
+import Shelley.Spec.Ledger.BlockChain
+import Shelley.Spec.Ledger.Tx
+import Cardano.Slotting.Block
+
+import Test.Shelley.Spec.Ledger.Generator.ScriptClass
+import Test.Shelley.Spec.Ledger.Generator.Core
+import Test.Shelley.Spec.Ledger.Generator.ShelleyEraGen ()
+import Test.Shelley.Spec.Ledger.Generator.Presets
+import Shelley.Spec.Ledger.OCert
+import Shelley.Spec.Ledger.Genesis
+import Shelley.Spec.Ledger.API.Genesis
+
+type KeyPair' crypto = (KeyPair 'Payment crypto, KeyPair 'Staking crypto)
+
+data ShelleyEraState crypto = ShelleyEraState
```

Comment:
I'm a bit concerned that this is tied to the Shelley era. Given this is basically model state, we shouldn't need it to be era specific.

Reply:
This was not well named and has been renamed.